kythe/extractors/openjdk11/java_wrapper/java_wrapper.go

```
@@ -0,0 +1,215 @@
+			result = append(result, a, v+",java.logging,java.sql")
+		case "--doclint-format":
+			_, args = shift(args)
+		case "-Werror":
+		default:
+			switch {
+			case strings.HasPrefix(a, "-Xplugin:depend"), strings.HasPrefix(a, "-Xlint:"), strings.HasPrefix(a, "-Xdoclint"):
+			case strings.HasPrefix(a, "-Xmx"):
+				result = append(result, "-Xmx3G")
+			default:
+				result = append(result, a)
+			}
+		}
+	}
+	// As we can only do anything meaningful with Java compilations,
+	// but wrap the java binary, don't attempt to exact other invocations.
+	if !isJavac {
+		return nil
+	}
+	return result
+}
+
+func extractorEnv() []string {
+	env := os.Environ()
+	env = setEnvDefault(env, kytheRootVar, ".")
+	env = setEnvDefault(env, kytheOutputVar, outputDir())
+	env = setEnvDefault(env, kytheTargetVar, moduleName())
+	env = setEnvDefault(env, kytheCorpusVar, "openjdk11")
+	return env
+}
+
+func setEnvDefault(env []string, key, def string) []string {
+	if val := os.Getenv(key); len(val) == 0 {
+		env = append(env, key+"="+def)
+	}
+	return env
+}
+
+func shift(args []string) (string, []string) {
+	if len(args) > 0 {
+		return args[0], args[1:]
+	}
+	return "", nil
+}
+
+func init() {
+	setupRunfiles()
+}
+
+func main() {
+	java, err := findJavaCommand()
+	if err != nil {
+		log.Fatalf("ERROR: unable to find java executable: %v", err)
+	}
+	jar, err := findExtractorJar()
+	if err != nil {
+		log.Fatalf("ERROR: unable to find extractor jar: %v", err)
+	}
+	if args := extractorArgs(os.Args[1:], jar); len(args) > 0 {
+		cmd := exec.Command(java, args...)
+		cmd.Env = extractorEnv()
+		log.Printf("*** Extracting: %s", moduleName())
+		if output, err := cmd.CombinedOutput(); err != nil {
+			w, err := os.Create(filepath.Join(outputDir(), moduleName()+".err"))
+			if err != nil {
+				log.Printf("Error creating error log for module %s: %v", moduleName(), err)
+			}
+			defer w.Close()
+			if _, err := fmt.Fprintf(w, "--- %s\n", shell.Join(args)); err != nil {
```

It's generally fine to ignore the error in a case like this where you have successfully opened the file for writing anyway and are just writing diagnostics. I would just drop the extra check/log wrappers on these.